data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/S3Service.java

```diff
@@ -5,55 +5,37 @@
 
 package org.opensearch.dataprepper.plugins.source;
 
-import org.opensearch.dataprepper.metrics.PluginMetrics;
-import org.opensearch.dataprepper.model.buffer.Buffer;
-import org.opensearch.dataprepper.model.event.Event;
-import org.opensearch.dataprepper.model.record.Record;
-import org.opensearch.dataprepper.plugins.source.codec.Codec;
-import org.opensearch.dataprepper.plugins.source.compression.CompressionEngine;
-import org.opensearch.dataprepper.plugins.source.ownership.BucketOwnerProvider;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
 import software.amazon.awssdk.core.retry.RetryPolicy;
 import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
 
 import java.io.IOException;
-import java.util.function.BiConsumer;
 
 public class S3Service {
     private static final Logger LOG = LoggerFactory.getLogger(S3Service.class);
 
     private final S3SourceConfig s3SourceConfig;
-    private final Buffer<Record<Event>> buffer;
-    private final S3Client s3Client;
-    private final Codec codec;
-    private final CompressionEngine compressionEngine;
-    private final PluginMetrics pluginMetrics;
-    private final BucketOwnerProvider bucketOwnerProvider;
-    private final S3ObjectWorker s3ObjectWorker;
+    private final S3ObjectHandler s3ObjectHandler;
 
-    S3Service(final S3SourceConfig s3SourceConfig,
-              final Buffer<Record<Event>> buffer,
-              final Codec codec,
-              final PluginMetrics pluginMetrics,
-              final BucketOwnerProvider bucketOwnerProvider) {
+    S3Service(final S3ObjectHandler s3ObjectHandler,
+    		  final S3SourceConfig s3SourceConfig) {
         this.s3SourceConfig = s3SourceConfig;
-        this.buffer = buffer;
-        this.codec = codec;
-        this.pluginMetrics = pluginMetrics;
-        this.bucketOwnerProvider = bucketOwnerProvider;
-        this.s3Client = createS3Client();
-        this.compressionEngine = s3SourceConfig.getCompression().getEngine();
-        final BiConsumer<Event, S3ObjectReference> eventMetadataModifier = new EventMetadataModifier(s3SourceConfig.getMetadataRootKey());
-        this.s3ObjectWorker = new S3ObjectWorker(s3Client, buffer, compressionEngine, codec, bucketOwnerProvider,
-                s3SourceConfig.getBufferTimeout(), s3SourceConfig.getNumberOfRecordsToAccumulate(), eventMetadataModifier, pluginMetrics);
+        this.s3ObjectHandler =s3ObjectHandler;
     }
 
     void addS3Object(final S3ObjectReference s3ObjectReference) {
-        try {
```

Comment:
With the refactoring you provided, you no longer need this condition. You can use the original implementation for the `addS3Object` method.